src/external-types/reveal.js/index.d.ts

```diff
@@ -4,9 +4,199 @@
+    // Loop the presentation
+    loop?: boolean,
+    // Number of slides away from the current that are visible on mobile
+    // devices. It is advisable to set this to a lower number than
+    // viewDistance in order to save resources
+    mobileViewDistance?: number,
+    // Enable slide navigation via mouse wheel
+    mouseWheel?: boolean,
+    // Changes the behavior of our navigation directions.
+    //
+    // "default"
+    // Left/right arrow keys step between horizontal slides, up/down
+    // arrow keys step between vertical slides. Space key steps through
+    // all slides (both horizontal and vertical).
+    //
+    // "linear"
+    // Removes the up/down arrows. Left/right arrows step through all
+    // slides (both horizontal and vertical).
+    //
+    // "grid"
+    // When this is enabled, stepping left/right from a vertical stack
+    // to an adjacent vertical stack will land you at the same vertical
+    // index.
+    //
+    // Consider a deck with six slides ordered in two vertical stacks:
+    // 1.1    2.1
+    // 1.2    2.2
+    // 1.3    2.3
+    //
+    // If you're on slide 1.3 and navigate right, you will normally move
+    // from 1.3 -> 2.1. If "grid" is used, the same navigation takes you
+    // from 1.3 -> 2.3.
+    navigationMode?: NavigationMode,
+    // Enable the slide overview mode
+    overview?: boolean,
+    // Flags if it should be possible to pause the presentation (blackout)
+    pause?: boolean,
+    // The maximum number of pages a single slide can expand onto when printing
+    // to PDF, unlimited by default
+    /* pdfMaxPagesPerSlide?: number, // ToDo: I think this is not needed */
+    /* pdfPageHeightOffset: -1, // ToDo: I think this is not needed */
+    /* pdfSeparateFragments: true, // ToDo: I think this is not needed */
+    // Exposes the reveal.js API through window.postMessage
+    /* postMessage?: boolean, // ToDo: I think this is not needed */
+    // Dispatches all reveal.js events to the parent window through postMessage
+    /* postMessageEvents?: boolean, // ToDo: I think this is not needed */
+    // Global override for preloading lazy-loaded iframes
+    // - null:   Iframes with data-src AND data-preload will be loaded when within
+    //           the viewDistance, iframes with only data-src will be loaded when visible
+    // - true:   All iframes with data-src will be loaded when within the viewDistance
+    // - false:  All iframes with data-src will be loaded only when visible
+    preloadIframes?: boolean | null,
+    // Opens links in an iframe preview overlay
+    // Add `data-preview-link` and `data-preview-link="false"` to customise each link
+    // individually
+    previewLinks?: boolean,
+    // Display a presentation progress bar
+    progress?: boolean,
+    // Flags if we should monitor the hash and change slides accordingly
+    /* respondToHashChanges?: boolean, // ToDo: I think this is not needed */
```

Comment:
> Flags if we should monitor the hash and change slides accordingly

I think this also depends on, if we want `hash` and therefore `fragmentInURL`…

Reply:
That becomes tricky with react-router, I guess.

Reply:
This won't work. Inside of the iframe it would only react to changes of the iframe URL.